src/duke/Parser.java

```diff
@@ -0,0 +1,97 @@
+package duke;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class Parser {
+    Parser() {}
+
+    /**
+     * Processes all the commands given by the user and triggers functions correspondingly.
+     * @param command command given by the user
+     * @param tl the TaskList object defined in Duke.
+     * @param ui the Ui object defined in Duke.
+     * @throws Exception
+     */
+    public static void processCommand(String command, TaskList tl, Ui ui) throws Exception {
+        int indexOfSlash = command.indexOf('/');
+        ArrayList<Task> arr = tl.getArr();
+        if (command.equals("bye")) {
+            ui.respondToBye();
+        } else if (command.length() >= 4) {
+            if (command.substring(0, 4).equals("list")) {
+                if (arr.size() == 0) {
+                    ui.respondToEmptyList();
+                } else {
+                    ui.respondToList();
+                }
+            } else if (command.substring(0, 4).equals("find")) {
+                if (command.length() == 4) {
+                    ui.respondToFindWrongSyntax();
+                } else {
+                    String searchWord = command.substring(5);
+                    ui.respondToFind(searchWord);
+                }
+            } else if (command.substring(0, 4).equals("done")) {
+                if (command.length() == 4) {
+                    ui.respondToDoneWrongSyntax();
+                } else {
+                    int taskNumber = Integer.parseInt(command.substring(5));
+                    if (taskNumber > arr.size()) {
+                        ui.respondToDoneFail();
+                    } else {
+                        ui.respondToDone(taskNumber);
+                    }
+                }
+            } else if (command.substring(0, 4).equals("todo")) {
+                if (command.length() == 4 || indexOfSlash != -1) {
+                    ui.respondToTodoWrongSyntax();
+                } else {
+                    ui.respondToTodo(command.substring(5));
+                }
+            } else if (command.substring(0, 5).equals("event")) {
+                if (indexOfSlash == -1 || command.length() == 5 || !command.substring(indexOfSlash + 1, indexOfSlash + 3).equals("at")) {
```

Comment:
Since this if statement has a long condition, perhaps you might want to display each new condition on a new line to improve readability?
I have also seen this case a few other times in your code.